packages/flutter/lib/src/cupertino/text_selection.dart

```diff
@@ -247,28 +247,38 @@ class CupertinoTextSelectionControls extends TextSelectionControls {
 
   /// Builder for iOS text selection edges.
   @override
-  Widget buildHandle(BuildContext context, TextSelectionHandleType type, double textLineHeight, [VoidCallback? onTap]) {
+  Widget buildHandle(BuildContext context, TextSelectionHandleType type, double textLineHeight, [VoidCallback? onTap, double? startGlyphHeight, double? endGlyphHeight]) {
     // iOS selection handles do not respond to taps.
 
     // We want a size that's a vertical line the height of the text plus a 18.0
     // padding in every direction that will constitute the selection drag area.
-    final Size desiredSize = getHandleSize(textLineHeight);
+    startGlyphHeight = startGlyphHeight ?? textLineHeight;
+    endGlyphHeight = endGlyphHeight ?? textLineHeight;
 
-    final Widget handle = SizedBox.fromSize(
-      size: desiredSize,
-      child: CustomPaint(
-        painter: _TextSelectionHandlePainter(CupertinoTheme.of(context).primaryColor),
-      ),
+    final Size desiredSize;
+    final Widget handle;
+
+    final Widget customPaint = CustomPaint(
+      painter: _TextSelectionHandlePainter(CupertinoTheme.of(context).primaryColor),
     );
 
     // [buildHandle]'s widget is positioned at the selection cursor's bottom
     // baseline. We transform the handle such that the SizedBox is superimposed
     // on top of the text selection endpoints.
     switch (type) {
       case TextSelectionHandleType.left:
+        desiredSize = getHandleSize(startGlyphHeight);
```

Comment:
Just want to double check that this still works in RTL mode (the left handle still corresponds to the "start" glyph height.

Reply:
Good catch, i'll try this out.

Reply:
https://github.com/flutter/flutter/issues/86763 Opened up a new issue. The start handle is broken only on iOS when switching text direction to RTL, it is unrelated to this change though I was able to reproduce on master.